# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: devtools/server/actors/watcher/target-helpers/utils.js
# Commit: 895cf08432c4
# Full Hash: 895cf08432c40fe371ff83e846cb161430c6be4b
# Author: Nicolas Chevobbe <nchevobbe@mozilla.com>
# Date: 2020-10-27 04:41:26
# Regressor Bug: 1633712
# File Overlap Count: 1
# Description:
#   Bug 1633712 - [devtools] Create WorkerTargets as soon as possible. r=ochameau,devtools-backward-compat-reviewers.
#   
#   This patch adds support for dedicated worker targets in the Watcher actor.
#   Shared and Service workers are not handled yet.
#   
# ==============================================================================

diff -r 2ce79a9ed2d5 -r 895cf08432c4 devtools/server/actors/watcher/target-helpers/utils.js
--- a/devtools/server/actors/watcher/target-helpers/utils.js	Mon Oct 26 12:52:26 2020 +0000
+++ b/devtools/server/actors/watcher/target-helpers/utils.js	Mon Oct 26 12:53:43 2020 +0000
@@ -7,12 +7,21 @@
 const Services = require("Services");
 
 /**
- * Helper function to know if a given WindowGlobal should be exposed via watchTargets("frame") API
+ * Helper function to know if a given WindowGlobal should be exposed via watchTargets API
  * XXX: We probably want to share this function with DevToolsFrameChild,
  * but may be not, it looks like the checks are really differents because WindowGlobalParent and WindowGlobalChild
  * expose very different attributes. (WindowGlobalChild exposes much less!)
+ *
+ * @param {BrowsingContext} browsingContext: The browsing context we want to check the window global for
+ * @param {String} watchedBrowserId
+ * @param {Object} options
+ * @param {Boolean} options.acceptNonRemoteFrame: Set to true to not restrict to remote frame only
  */
-function shouldNotifyWindowGlobal(browsingContext, watchedBrowserId) {
+function shouldNotifyWindowGlobal(
+  browsingContext,
+  watchedBrowserId,
+  options = {}
+) {
   const windowGlobal = browsingContext.currentWindowGlobal;
   // Loading or destroying BrowsingContext won't have any associated WindowGlobal.
   // Ignore them. They should be either handled via DOMWindowCreated event or JSWindowActor destroy
@@ -42,7 +51,11 @@
     return false;
   }
 
-  // For now, we only mention the "remote frames".
+  if (options.acceptNonRemoteFrame) {
+    return true;
+  }
+
+  // If `acceptNonRemoteFrame` options isn't true, only mention the "remote frames".
   // i.e. the frames which are in a distinct process compared to their parent document
   return (
     !browsingContext.parent ||